server/text_generation_server/models/custom_modeling/mllama.py

```diff
@@ -710,34 +710,41 @@ def forward(
         # )
         if SYSTEM == "ipex":
             attn_output = torch.empty_like(query_states)
-            ipex.llm.functional.varlen_attention(
-                (
-                    query_states.contiguous()
-                    if query_states.device.type == "xpu"
-                    else query_states
-                ),
-                (
-                    key_states.contiguous()
-                    if key_states.device.type == "xpu"
-                    else key_states
-                ),
-                (
-                    value_states.contiguous()
-                    if value_states.device.type == "xpu"
-                    else value_states
-                ),
-                attn_output,
-                cu_seqlen_q,
-                cu_seqlen_k,
-                max_q,
-                max_k,
-                0.0,
-                self.softmax_scale,
-                False,
-                causal,
-                False,
-                None,
-            )
+            if query_states.device.type == "xpu":
+                ipex.llm.functional.varlen_attention(
+                    query_states.contiguous(),
+                    key_states.contiguous(),
+                    value_states.contiguous(),
+                    attn_output,
+                    cu_seqlen_q,
+                    cu_seqlen_k,
+                    None,
```

Comment:
there's a None here but not in the cpu case is that normal ? I would assume the function has different logic on cpu & xpu, but is the signature different as well ?

Reply:
we has fix in next ipex rls, for 2.7, just a WA, or else xpu will crash